src/main/java/hexgui/gui/HexGui.java

```diff
@@ -0,0 +1,2393 @@
+import hexgui.htp.HtpError;
+import hexgui.sgf.SgfReader;
+import hexgui.sgf.SgfWriter;
+import hexgui.util.ErrorMessage;
+import hexgui.util.Pair;
+import hexgui.util.StreamCopy;
+import hexgui.util.StringUtils;
+import hexgui.version.Version;
+import java.awt.*;
+import java.awt.event.*;
+import java.io.*;
+import java.net.*;
+import java.util.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.Semaphore;
+import javax.swing.*;
+
+// ----------------------------------------------------------------------------
+
+/** HexGui. */
+public final class HexGui extends JFrame
+    implements ActionListener,
+        GuiBoard.Listener,
+        HtpShell.Callback,
+        HtpController.GuiFxCallback,
+        AnalyzeDialog.Listener,
+        Comment.Listener {
+  public HexGui(final File file, final String command) {
+    super("HexGui");
+    setIcon();
+
+    System.out.println("HexGui v" + Version.id + "; " + Version.date + "\n");
+
+    // Catch the close action and shutdown nicely
+    setDefaultCloseOperation(JFrame.DO_NOTHING_ON_CLOSE);
+    addWindowListener(
+        new java.awt.event.WindowAdapter() {
+          public void windowClosing(WindowEvent winEvt) {
+            cmdShutdown();
+          }
+        });
+
+    m_selected_cells = new Vector<HexPoint>();
+
+    m_about = new AboutDialog(this);
+
+    m_preferences = new GuiPreferences(getClass());
+
+    m_menubar = new GuiMenuBar(this, m_preferences);
+    setJMenuBar(m_menubar.getJMenuBar());
+
+    m_toolbar = new GuiToolBar(this, m_preferences);
+    getContentPane().add(m_toolbar.getJToolBar(), BorderLayout.NORTH);
+
+    m_statusbar = new StatusBar();
+    getContentPane().add(m_statusbar, BorderLayout.SOUTH);
+
+    m_guiboard = new GuiBoard(this, m_preferences);
+    getContentPane().add(m_guiboard, BorderLayout.CENTER);
+
+    m_showAnalyzeText = new ShowAnalyzeText(this, m_guiboard);
+
+    JPanel panel = new JPanel(new BorderLayout());
+    getContentPane().add(panel, BorderLayout.EAST);
+
+    m_blackClock = new Clock();
+    m_whiteClock = new Clock();
+    m_gameinfopanel = new GameInfoPanel(m_blackClock, m_whiteClock);
+    m_comment = new Comment(this);
+    panel.add(m_gameinfopanel, BorderLayout.NORTH);
+    panel.add(m_comment, BorderLayout.CENTER);
+
+    cmdNewGame();
+
+    pack();
+
+    m_locked = false;
+
+    m_semaphore = new Semaphore(1);
+    m_htp_queue = new ArrayBlockingQueue<HtpCommand>(256);
+    new Thread(new CommandHandler(this, m_htp_queue)).start();
+
+    setVisible(true);
+    // After frame is visible, further code using Swing functions must
+    // be run in the Swing event dispatch thread.
+    SwingUtilities.invokeLater(
+        new Runnable() {
+          public void run() {
+            initialize(file, command);
+          }
+        });
+    setCursorType();
+  }
+
+  // -------------------------------------------------------------------
+
+  public void actionPerformed(ActionEvent e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 54. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.